src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -364,6 +364,274 @@ CREATE NONCLUSTERED INDEX IX_ChangeFeed_StudyInstanceUid_SeriesInstanceUid_SopIn
+    Custom Tag Data Table for VR Types mapping to Double
+**************************************************************/
+CREATE TABLE dbo.CustomTagDouble (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                FLOAT(53)            NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
+    Watermark               BIGINT               NOT NULL
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTagDouble ON dbo.CustomTagDouble
+(
+    TagKey,
+    StudyKey,
+    SeriesKey,
+    InstanceKey
+)
+INCLUDE
+(
+    WaterMark
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDouble_TagKey_StudyKey ON dbo.CustomTagDouble
+(
+    TagKey,
+    StudyKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDouble_TagKey_SeriesKey ON dbo.CustomTagDouble
+(
+    TagKey,
+    SeriesKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDouble_TagKey_InstanceKey ON dbo.CustomTagDouble
+(
+    TagKey,
+    InstanceKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDouble_TagKey_TagValue ON dbo.CustomTagDouble
+(
+    TagKey,
+    TagValue
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to DateTime
+**************************************************************/
+CREATE TABLE dbo.CustomTagDateTime (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                DATETIME2(7)         NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
+    Watermark               BIGINT               NOT NULL
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTagDateTime ON dbo.CustomTagDateTime
+(
+    TagKey,
+    StudyKey,
+    SeriesKey,
+    InstanceKey
+)
+INCLUDE
+(
+    WaterMark
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDateTime_TagKey_StudyKey ON dbo.CustomTagDateTime
+(
+    TagKey,
+    StudyKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDateTime_TagKey_SeriesKey ON dbo.CustomTagDateTime
+(
+    TagKey,
+    SeriesKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDateTime_TagKey_InstanceKey ON dbo.CustomTagDateTime
+(
+    TagKey,
+    InstanceKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDateTime_TagKey_TagValue ON dbo.CustomTagDateTime
+(
+    TagKey,
+    TagValue
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to PersonName
+**************************************************************/
+CREATE TABLE dbo.CustomTagPersonName (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                NVARCHAR(200)        COLLATE SQL_Latin1_General_CP1_CI_AI NOT NULL,
```

Comment:
how about NVARCHAR(324)?
As DICOM  standard, the possible max name length should be 324, I'll find the link later.

Reply:
http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html

There are totally 5 component group , and max length of each is 64, plus 4 delimiters, add up to 324

Reply:
It seems like our current table for PersonName only uses 200. Let me check and see why that works!

---
In reply to: [565726566](https://github.com/microsoft/dicom-server/pull/523#discussion_r565726566) [](ancestors = 565726566)

Reply:
I brought up this question before in standup, and created user story https://microsofthealth.visualstudio.com/Health/_workitems/edit/79257. But we could double check Smitha who wrote this SQL before.